sklearn/model_selection/tests/test_search.py

```diff
@@ -929,6 +961,109 @@ def test_search_iid_param():
         assert_almost_equal(train_std, 0)
 
 
+def test_grid_search_cv_results_multimetric():
+    X, y = make_classification(n_samples=50, n_features=4, random_state=42)
+
+    n_splits = 3
+    params = [dict(kernel=['rbf', ], C=[1, 10], gamma=[0.1, 1]),
+              dict(kernel=['poly', ], degree=[1, 2])]
+
+    for iid in (False, True):
+        grid_searches = []
+        for scoring in ({'accuracy': make_scorer(accuracy_score),
+                         'recall': make_scorer(recall_score)},
+                        'accuracy', 'recall'):
+            grid_search = GridSearchCV(SVC(), cv=n_splits, iid=iid,
+                                       param_grid=params, scoring=scoring,
+                                       refit=False)
+            grid_search.fit(X, y)
+            assert_equal(grid_search.iid, iid)
+            grid_searches.append(grid_search)
+
+        compare_cv_results_multimetric_with_single_metric_accuracy_recall(
+            *grid_searches, iid=iid)
+
+
+def test_random_search_cv_results_multimetric():
+    X, y = make_classification(n_samples=50, n_features=4, random_state=42)
+
+    n_splits = 3
+    n_search_iter = 30
+    scoring = ('accuracy', 'recall')
+
+    # Scipy 0.12's stats dists do not accept seed, hence we use param grid
+    params = dict(C=np.logspace(-10, 1), gamma=np.logspace(-5, 0, base=0.1))
+    for iid in (True, False):
+        for refit in (True, False):
+            random_searches = []
+            for scoring in (('accuracy', 'recall'), 'accuracy', 'recall'):
+                # If True, for multi-metric pass refit='accuracy'
+                if refit:
+                    refit = 'accuracy' if isinstance(scoring, tuple) else refit
+                clf = SVC(probability=True, random_state=42)
+                random_search = RandomizedSearchCV(clf, n_iter=n_search_iter,
+                                                   cv=n_splits, iid=iid,
+                                                   param_distributions=params,
+                                                   scoring=scoring,
+                                                   refit=refit, random_state=0)
+                random_search.fit(X, y)
+                random_searches.append(random_search)
+
+            compare_cv_results_multimetric_with_single_metric_accuracy_recall(
+                *random_searches, iid=iid)
+            if refit:
+                compare_refit_methods_when_refit_with_acc(
+                    random_searches[0], random_searches[1], refit)
+
+
+def compare_cv_results_multimetric_with_single_metric_accuracy_recall(
```

Comment:
Many of your function names are too verbose. Something like "accuracy_recall" can be described in a comment. `compare_cv_results_multimetric_with_single` is sufficient; `check_multimetric_cv_results` is better.